2-mandatory/2-oxygen-levels.js

```diff
@@ -11,7 +11,26 @@
     Some string methods that might help you here are .replace() and .substring().
 */
 
-function findSafeOxygenLevel() {}
+//ERROR --->>>> indSafeOxygenLevel function returns undefined if no valid planets found
+
+function findSafeOxygenLevel(array) {
+  if (array.some((number) => number.includes("%"))) {
+    let arrayWithPercentegOnly = array.filter((number) => number.includes("%"));
+    let convertedToNumbers = arrayWithPercentegOnly.map((number) =>
+      parseFloat(number)
+    );
+    let result = convertedToNumbers.find(checkLevel) + "%";
+    return result;
+  }
+
+  function checkLevel(number) {
+    if (number > 19.5 && number < 23.5) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+}
```

Comment:
You could make this simpler by:
```suggestion
  function checkLevel(number) {
    return (number > 19.5 && number < 23.5);
}
```